### PR TITLE
Test AWS 2 S3 properly

### PR DESCRIPTION
**Pull request by ppalaga**
@llowinge you may want to review

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
Well done. We should do the same for all the aws components.

File: integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3IT.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.aws2;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.NativeImageTest;
+
+@NativeImageTest
+@QuarkusTestResource(AwsTestResource.class)
```
Comment (aldettinger) [inline integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3IT.java:23]: Maybe this one is already inherited from Aws2S3Test ?

File: integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3Test.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                .then()
+                .statusCode(201);
+
+        // Read
+        RestAssured.get("/aws2/s3/object/" + oid)
+                .then()
+                .statusCode(200)
+                .body(is(blobContent));
+
+        // Read
```
Comment (aldettinger) [inline integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3Test.java:64]: Any reason to read two times in a row ?

**Review by aldettinger — APPROVED** [review]
Two minor questions, but looks very good overall :+1: localstack is a nice spot.

File: integration-tests/aws2-s3/pom.xml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-main</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-aws2-s3</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy</artifactId>
```
Comment (zbendhiba) [inline integration-tests/aws2-s3/pom.xml:43]: Do we need quarkus-resteasy and quarkus-resteasy-jackson?

File: integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3IT.java (hunk 1 above)
Comment (ppalaga) [reply integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3IT.java:23]: I always thought, it must be present in both classes. You make me wonder 🤔

File: integration-tests/aws2-s3/pom.xml (hunk 3 above)
Comment (ppalaga) [reply integration-tests/aws2-s3/pom.xml:43]: Yeah, quarkus-resteasy-jackson pulls quarkus-resteasy IIRC, so yes, quarkus-resteasy could be omitted.

File: integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3Test.java (hunk 2 above)
Comment (ppalaga) [reply integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3Test.java:64]: Good catch, thanks! It is a leftover from some intermediate experiments.

File: integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3IT.java (hunk 1 above)
Comment (ppalaga) [reply integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3IT.java]: Indeed. I learned something new, thanks! :)

Comment (ppalaga) [reply integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3IT.java]: Fixed in 13abcff

File: integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3Test.java (hunk 2 above)
Comment (ppalaga) [reply integration-tests/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/Aws2S3Test.java]: Fixed in 13abcff

File: integration-tests/aws2-s3/pom.xml (hunk 3 above)
Comment (ppalaga) [reply integration-tests/aws2-s3/pom.xml:43]: No, quarkus-resteasy-jackson does not depend on quarkus-resteasy, so they are both necessary.

**Comment by ppalaga** [comment]
13abcff addressed all addressable comments.

**Pull request merged**